### PR TITLE
Add various JavaScript features

### PR DESCRIPTION
**Pull request by Elchi3**
Many of these things should stand on their own, so here are various JS features at once.

### CODE REVIEW

File: features/top-level-await.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+name: Top-level await
+description: Use the `await` keyword outside of async functions and at the top level of modules.
```
Comment (captainbrosset) [inline features/top-level-await.yml:2]: "outside of async functions and at the top level of modules" sounds a little bit (to me) like there might be other places where top-level await can be used.


```suggestion
description: The `await` keyword, when used at the top level of a module (outside of an async function), delays parent module execution until after a promise fulfills.
```

**Review by captainbrosset — APPROVED** [review]

File: features/trailing-commas-in-functions.yml
Comment (captainbrosset) [inline features/trailing-commas-in-functions.yml:1]: I'm tempted to remove this feature, and put the key inside the functions feature instead (ignored, using a compute_from).

File: features/optional-catch-binding.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+name: Optional catch binding
+description: Omits the `catch` binding in the `try...catch` statement.
```
Comment (captainbrosset) [inline features/optional-catch-binding.yml:2]:
```suggestion
description: Omit the the binding parameter of a `catch` clause when you don't need information about the exception in a `try...catch` statement.
```

File: features/nullish-coalescing.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+name: Nullish coalescing operators
```
[the remark this answers is not in the document]
Comment (Elchi3) [reply features/nullish-coalescing.yml:1]:
```suggestion
name: Nullish coalescing
```

File: features/trailing-commas-in-functions.yml
Comment (Elchi3) [reply features/trailing-commas-in-functions.yml:1]: That makes total sense and the function feature already has other keys regarding trailing commas. Nice!

File: features/let-const.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+name: Let and const
+description: The `let` and `const` declarations define block-scoped variables.
+spec: https://tc39.es/ecma262/multipage/ecmascript-language-statements-and-declarations.html#sec-let-and-const-declarations
```
Comment (ddbeck) [inline features/let-const.yml:3]: I see we've come around on https://github.com/web-platform-dx/web-features/pull/330 😂 

```suggestion
spec: https://tc39.es/ecma262/multipage/ecmascript-language-statements-and-declarations.html#sec-let-and-const-declarations
caniuse:
  - let
  - const
```

File: features/logical-assignments.yml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+name: Logical assignments
+description: The logical AND assignment (&&=) and the logical OR assignment (||=) operators short-circuit the respective binary logical operators.
```
Comment (ddbeck) [inline features/logical-assignments.yml:2]:
```suggestion
description: The logical AND assignment (`&&=`) and the logical OR assignment (`||=`) operators short-circuit the respective binary logical operators.
```

File: features/numeric-seperators.yml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+name: Numeric separators
+description: "To improve readability for numeric literals, underscores (`_`) can be used as separators. For example: `1_050.95`"
```
Comment (ddbeck) [inline features/numeric-seperators.yml:2]:
```suggestion
description: To improve readability for numeric literals, underscores (`_`) can be used as separators. For example, `1_050.95` is equivalent to `1050.95`.
```

File: features/optional-catch-binding.yml [hunk 7]
```diff
@@ -0,0 +1,7 @@
+name: Optional catch binding
+description: Omit the the binding parameter of a `catch` clause when you don't need information about the exception in a `try...catch` statement.
```
Comment (ddbeck) [inline features/optional-catch-binding.yml:2]: We've offset `…` as a placeholder with spaces elsewhere 

```suggestion
description: Omit the the binding parameter of a `catch` clause when you don't need information about the exception in a `try ... catch` statement.
```

File: features/template-literals.yml [hunk 8]
```diff
@@ -0,0 +1,7 @@
+name: Template literals
+description: Template literals are literals delimited with backtick (`) characters, allowing for multi-line strings, string interpolation with embedded expressions, and special constructs called tagged templates.
```
Comment (ddbeck) [inline features/template-literals.yml:2]: For clarity, to show that we mean the character alone and that this is not some markdown issue.

```suggestion
description: Template literals are literals delimited with backtick (<code>`</code>) characters, allowing for multi-line strings, string interpolation with embedded expressions, and special constructs called tagged templates.
```

**Review by ddbeck — APPROVED** [review]
I have one substantive suggestion (about a caniuse link), plus some totally optional description nits. But I think you should merge this soon after you've looked this over.

**Pull request merged**